crawl-ref/source/item-use.cc

```diff
@@ -1708,13 +1709,16 @@ static bool _puton_item(int item_slot, bool prompt_slot)
 
     // It looks to be possible to equip this item. Before going any further,
     // we should prompt the user with any warnings that come with trying to
-    // put it on.
-    if (!check_warning_inscriptions(item, OPER_PUTON))
+    // put it on, except when they have already been prompted with them
+    // from switching rings.
+    if (check_for_inscriptions)
```

Comment:
These could probably be combined into a single `if (... && ...)`

Reply:
I am not 100% sure how C++ works so feel free to correct me. I assumed that if it was all in one `if(...)`, the function check_warning_inscriptions would be run even when check_for_inscriptions was false, and it's this function which causes the prompt -- we do not want to call this function if check_for_inscriptions is false.